### DISCUSSION TITLE
How to limit TEXT field type length in SQLite database backend?

### DISCUSSION THREAD

**Discussion by MurzNN** [prisma/prisma#7082] [Q&A]
On SQLite there is ability to limit the length of TEXT field type via passing numbers in brackets of field definition line (`VARCHAR(32)` or `CHAR(2)`). But I can't find the way to pass this via Prisma Schema, can anybody show me how to do this?
For PostgreSQL this works via `VarChar` and `Char`, something like this:
```
charfield      String?    @postgresql.Char(2)
```
And same for MySQL.
But not working for SQLite. Is this not yet implemented (does the issue exist to track?) or there are some limitations exists?

**Comment by surfmuggle** [comment]
To my understanding of [Datatypes In SQLite](https://www.sqlite.org/datatype3.html) and this bit

> If the declared type of the column contains any of the strings "CHAR", "CLOB", or "TEXT" then that column has TEXT affinity. Notice that the type VARCHAR contains the string "CHAR" and is thus assigned TEXT affinity.

SQLite does not support any restrictions on a column of type `TEXT`. The type `VARCHAR(3)` does not exist natively and will be converted to a matching type based of the [column affinity](https://www.sqlite.org/datatype3.html#determination_of_column_affinity). 

You can try it out like this

```SQL
CREATE TABLE IF NOT EXISTS  "TextColumn" (
	"Id"	INTEGER NOT NULL UNIQUE,	
	"A_3"	VARCHAR(3),
	"B_Text"	TEXT,
	PRIMARY KEY("Id" AUTOINCREMENT)
);
INSERT INTO TextColumn (A_3, B_Text)
VALUES
    ('ABC', 'ABC is a valid string for column A_3'),
    ('123', '123 is a valid string for column A_3'),
    ('ABC123', 'ABC123 is not a valid string for column A_3');
SELECT * FROM TextColumn;
```
See the output without any errors

![image](https://github.com/prisma/prisma/assets/1271555/1f870f44-2f4d-4437-8fb4-d13255b073f0)
